dm/pkg/checker/table_structure.go

```diff
@@ -149,8 +203,8 @@ func (c *TablesChecker) Name() string {
 	return "table structure compatibility check"
 }
 
-func (c *TablesChecker) checkCreateSQL(ctx context.Context, statement string) []*incompatibilityOption {
-	parser2, err := dbutil.GetParserForDB(ctx, c.db)
+func (c *TablesChecker) checkCreateSQL(ctx context.Context, db *sql.DB, statement string) []*incompatibilityOption {
+	parser2, err := dbutil.GetParserForDB(ctx, db)
```

Comment:
Maybe be we can prefetch to sql-mode and cache it with db to avoid fetch the sql-mode for every table